src/components/overlays/StartOverlay.tsx

```diff
@@ -0,0 +1,109 @@
+import React from "react";
+import { observer } from "mobx-react";
+import { Portal } from "./Portal";
+import styles from "./StartOverlay.module.scss";
+import { useMainStore } from "../../hooks/useMainStore";
+
+export const StartOverlay = observer(() => {
+  const { game, player } = useMainStore();
+  const [name, setName] = React.useState("");
+  const [money, setMoney] = React.useState(0);
+  const [formErrorMessage, setFormErrorMessage] = React.useState("");
+
+  const handleName = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setName(event.target.value.trim());
+  };
+
+  const handleMoney = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setMoney(+event.target.value);
+  };
+
+  const startGame = () => {
+    if (name !== "" && money > 0) {
+      game.nextRound();
+      player.setPlayerName(name);
+      player.addPlayerMoney(money);
```

Comment:
Probably you'd want to create extra method to set both **name** and **initial money** simultaneously and a separate method to update player's money